source/effect_codegen_glsl.cpp

```diff
@@ -205,18 +205,26 @@ class codegen_glsl final : public codegen
 
 	id make_id() { return _next_id++; }
 
-	template <bool is_function = false>
 	std::string id_to_name(id id) const
 	{
-		if constexpr (!is_function)
-			if (const auto it = _remapped_sampler_variables.find(id); it != _remapped_sampler_variables.end())
-				id = it->second;
+		if (const auto it = _remapped_sampler_variables.find(id); it != _remapped_sampler_variables.end())
+			id = it->second;
 		assert(id != 0);
 		if (const auto it = _names.find(id); it != _names.end())
-			return is_function ? it->second : it->second + '_' + std::to_string(id);
+			return it->second;
 		return '_' + std::to_string(id);
 	}
 
+	template <bool is_unique = true>
+	std::string &define_name(const id id, const std::string &name)
```

Comment:
The `name` parameter should best be a `std::string` instead of `const std::string &`, so that the move operator can work properly.

Reply:
Also you only call `define_name` once now. All other sites still use the `_names` map directly. Those occurances should probably replaced with a call to this function now too.